### ISSUE TITLE
Make Controls:TextBoxHelper.UseFloatingWatermark="True" work for DatePicker

### ISSUE THREAD

**Issue by VeegaP** [MahApps/MahApps.Metro#2269]
It would be cool if the option in TextBoxHelper.UseFloatingWatermark worked on DatePicker. Any way to make it work in case you don't plan to make it a functionality?
![floatingwatermark](https://cloud.githubusercontent.com/assets/8075379/11763097/a53eb5d6-a0fd-11e5-88ed-11cb985b8213.png)

As a temporary workaround I created a simple style to achieve what I'd like:
```xaml
<Style TargetType="{x:Type DatePickerTextBox}"
                   BasedOn="{StaticResource {x:Type DatePickerTextBox}}">
                <Setter Property="Template">
                    <Setter.Value>
                        <ControlTemplate>
                            <TextBox Controls:TextBoxHelper.Watermark="{Binding Path=(Controls:TextBoxHelper.Watermark),RelativeSource={RelativeSource TemplatedParent}}"
                                     Controls:TextBoxHelper.UseFloatingWatermark="True"
                                     VerticalContentAlignment="Center"
                                     Text="{Binding Text, RelativeSource={RelativeSource TemplatedParent}, Mode=TwoWay}"
                                     BorderThickness="0"
                                     Padding="0" />
                        </ControlTemplate>
                    </Setter.Value>
                </Setter>
            </Style>
```
![result](https://cloud.githubusercontent.com/assets/8075379/11763212/535b707e-a102-11e5-8cbe-3fdcd4c13060.png)

**Comment by VeegaP** [comment]
![floatingwatermark](https://cloud.githubusercontent.com/assets/8075379/11763097/a53eb5d6-a0fd-11e5-88ed-11cb985b8213.png)

**Comment by punker76** [comment]
@VeegaP available now after this 94b2a090acb53a74f5b84f84f1c11e1c8df62e8a
